Add Commit Signing section (#2019)
lambdaclass/cairo-vm: CONTRIBUTING.md

- You may want to **include screenshots and animated GIFs** which help you demonstrate the steps or point out the part which the suggestion is related to. You can use [this tool](https://www.cockos.com/licecap/) to record GIFs on macOS and Windows, and [this tool](https://github.com/colinkeenan/silentcast) or [this tool](https://github.com/GNOME/byzanz) on Linux. 
- **Explain why this enhancement would be useful** to most cairo-vm users. You may also want to point out the other projects that solved it better and which could serve as inspiration.

### Pushing commits to Pull Requests

> All commits of Pull Requests to this repo must be signed. See [this page](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits) for more details on how to sign a commit and how to manage signing keys in Github.
> 

Reviewer: Does the restriction apply to all pull requests or only to internal branches? If only to internal branches, we could phrase it as:
```suggestion
This repository enforces commit signing on all branches. See [this page](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits) for more details on how to sign a commit and how to manage signing keys in Github.
```
